modules/telemetry/include/hephaestus/telemetry/metric_record.h

```diff
@@ -0,0 +1,51 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+#include <string_view>
+
+#include "hephaestus/serdes/json.h"
+#include "hephaestus/telemetry/metric_sink.h"
+
+namespace heph::telemetry {
+namespace internal {
+[[nodiscard]] auto
+jsonToValuesMap(std::string_view json) -> std::unordered_map<std::string, Metric::ValueType>;
+}
+
+/// @brief Register a new data point sink.
+/// For every data point recorded, the sink will be called to consume the data.
+/// There is no limit on the number of sinks supported.
+void registerMetricSink(std::unique_ptr<IMetricSink> sink);
```

Comment:
as you write it here even in the comment, why not

```suggestion
/// @brief Register a new data point sink.
/// For every data point recorded, the sink will be called to consume the data.
/// There is no limit on the number of sinks supported.
void registerDatapointSink(std::unique_ptr<IDatapointSink> sink);
```